client/gpgcheck.c

```diff
@@ -376,3 +375,42 @@ VerifyRpmSig(
     }
     goto cleanup;
 }
+
+uint32_t
+TDNFImportGPGKey(
+    rpmts pTS,
+    const char* pszFile
+    )
+{
+    uint32_t dwError = 0;
+    pgpArmor nArmor = PGPARMOR_NONE;
+    uint8_t* pPkt = NULL;
+    size_t nPktLen = 0;
+    char* pszKeyData = NULL;
+
+    if(pTS == NULL || IsNullOrEmptyString(pszFile))
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = ReadGPGKey(pszFile, &pszKeyData);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    nArmor = pgpParsePkts(pszKeyData, &pPkt, &nPktLen);
+    if(nArmor != PGPARMOR_PUBKEY)
+    {
+        dwError = ERROR_TDNF_INVALID_PUBKEY_FILE;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = rpmtsImportPubkey(pTS, pPkt, nPktLen);
```

Comment:
Can't we add rpmtsImportPubkey() to AddKeyToKeyRing() instead of handling it in a separate function? Is this separately handled only because we do not have pTS as a parameter in TDNFGPGCheck() function?

Reply:
I think instead we can eliminate the `AddKeyToKeyRing()` function. It looks like rpmtsImportPubkey() already adds the key to he keyring (see https://github.com/rpm-software-management/rpm/blob/853c48ba6468ce1a516621a2fa6d1fc51e4f7410/lib/rpmts.c#L557).

How about doing this in another change?

Reply:
Yes, we can take it in a separate PR.